macros/models/_sector/tokens/balances_enrich.sql

```diff
@@ -1,47 +1,34 @@
-{%- macro balances_enrich(balances_base, blockchain, daily=false) %}
+{%- macro balances_enrich(balances_raw, daily=false) %}
 select
+    balances.blockchain,
     {% if daily %}balances.day,{% endif %}
     balances.block_number,
     balances.block_time,
-    balances.type,
-    balances.address as wallet_address,
-    balances.contract_address as token_address,
-    balances.amount as balance_raw,
+    balances.address,
+    balances.token_address,
+    balances.token_standard,
+    balances.balance_raw,
     CASE
-        WHEN balances.type = 'erc20' THEN balances.amount / power(10, erc20_tokens.decimals)
-        WHEN balances.type = 'native' THEN balances.amount / power(10, 18)
-        ELSE balances.amount
+        WHEN balances.token_standard = 'erc20' THEN balances.balance_raw / power(10, erc20_tokens.decimals)
+        WHEN balances.token_standard = 'native' THEN balances.balance_raw / power(10, 18)
+        ELSE balances.balance_raw
     END as balance,
-    CASE
-        WHEN balances.type = 'erc20' THEN balances.amount / power(10, erc20_tokens.decimals) * prices.price
-        WHEN balances.type = 'native' THEN balances.amount / power(10, 18) * prices.price
-        ELSE NULL
-    END as balance_usd,
-    prices.price as price_rate,
-    erc20_tokens.symbol,
-    erc20_tokens.decimals,
+    erc20_tokens.symbol as token_symbol,
     token_id,
     nft_tokens.name as collection_name
-from   {{ source('prices', 'usd') }} prices
-right join {{ balances_base }} balances on (
-    CASE
-        WHEN type = 'erc20' THEN prices.contract_address = balances.contract_address and prices.blockchain = '{{ blockchain }}'
-        WHEN type = 'native' THEN prices.contract_address is null and prices.symbol = 'ETH' and prices.blockchain is null
-        ELSE false
-    END)
-    and prices.minute = date_trunc('minute', balances.block_time)
+from {{balances_raw}} balances
 left join {{ source('tokens', 'erc20') }} erc20_tokens on
-    erc20_tokens.blockchain = '{{ blockchain }}' AND (
+    erc20_tokens.blockchain = balances.blockchain AND (
     CASE
-        WHEN type = 'erc20' THEN erc20_tokens.contract_address = balances.contract_address
+        WHEN token_standard = 'erc20' THEN erc20_tokens.contract_address = balances.token_address
         -- TODO: should not be hardcoded
-        WHEN type = 'native' THEN erc20_tokens.contract_address = 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
+        WHEN token_standard = 'native' THEN erc20_tokens.contract_address = 0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee
```

Comment:
This will not work for Polygon I think, we should pass this in instead.

Reply:
I have only changed the column name here.

Reply:
We use this for polygon https://github.com/duneanalytics/spellbook/blob/main/tokens/models/tokens/polygon/tokens_polygon_erc20.sql#L14

Most other L2s use ETH, we could actually also just pass in the the symbol and the decimal (it seems to be 8 for BNB?)